src/main/daml/Daml/Finance/Settlement/Instruction.daml

```diff
@@ -296,13 +286,15 @@ mustAuthorizeHelper authorizers parties =
     parties `S.isSubsetOf` authorizers
 
 -- | HIDE
-discloseAccountHelper : ((Text, Parties) -> Party -> Parties -> AccountKey -> Update a) ->
-  (Text, Parties) -> AccountKey -> Bool -> Update a
-discloseAccountHelper f (context, settlers) account isInstructor = do
-  viewA <- view <$> fetchInterfaceByKey @Account.R @Account.I account
-  let
-    disclosers = if isInstructor
-                 then viewA.controllers.outgoing
-                 else viewA.controllers.incoming
-    viewer = L.head $ S.toList disclosers
-  f (context, settlers) viewer disclosers account
+discloseAccountHelper : ((Text, Parties) -> Party -> Parties -> AccountKey -> Update a) -> [Party]
+  -> (Text, Parties) -> AccountKey  -> Parties -> Update a
+discloseAccountHelper f signatories (context, settlers) accountKey actors = do
```

Comment:
Why do we need `signatories` as an input?

If they are not the signatories of the `account`, can you use a more descriptive name e.g. `instructionSignatories`?

Reply:
Also, is there a more descriptive name for `f`?

Reply:
I am trying to understand if there is any business logic change in the `refactor` commits

Reply:
I changed to `instructionSignatories`, and `discloseAction`